tests/unit/test_charm.py

```diff
@@ -487,31 +488,33 @@ def test_update_database(
     """
     harness.set_can_connect(harness.model.unit.containers["wordpress"], True)
     harness.begin_with_initial_hooks()
+    patch.container._fail_wp_update_database = False
     charm: WordpressCharm = typing.cast(WordpressCharm, harness.charm)
     charm._on_update_database_action(action_event_mock)
 
     action_event_mock.set_results.assert_called_once_with({"result": "ok"})
     action_event_mock.fail.assert_not_called()
 
 
-def test_update_database_dry_run(
+def test_update_database_fail(
+    patch,
     harness: ops.testing.Harness,
     action_event_mock: unittest.mock.MagicMock,
 ):
     """
-    arrange: after charm is initialized and database ready.
-    act: run update-database action with dry-run.
-    assert: update-database action should success and return "ok".
+    arrange: after charm is initialized and database is not ready.
```

Comment:
Perhaps more explicitly: "and the database update is mocked to fail".

Reply:
Thanks! Done.